assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -140,12 +160,15 @@ export function registerDefaults( notificationsAPI ) {
 				resolveSelect,
 				dispatch,
 			} ) => {
-				const adSenseModuleConnected =
-					select( CORE_MODULES ).isModuleConnected( 'adsense' );
+				const adSenseModuleConnected = await resolveSelect(
+					CORE_MODULES
+				).isModuleConnected( 'adsense' );
 
-				const analyticsModuleConnected =
-					select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
+				const analyticsModuleConnected = await resolveSelect(
+					CORE_MODULES
+				).isModuleConnected( 'analytics-4' );
 
+				await resolveSelect( MODULES_ANALYTICS_4 ).getSettings();
```

Comment:
Here we can add `Promise.all` with `getModules` and `getSettings` calls to make it resolve faster.